controllers/provider-aws/pkg/controller/worker/machines.go

```diff
@@ -0,0 +1,183 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package worker
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	awsapi "github.com/gardener/gardener-extensions/controllers/provider-aws/pkg/apis/aws"
+	awsapihelper "github.com/gardener/gardener-extensions/controllers/provider-aws/pkg/apis/aws/helper"
+	confighelper "github.com/gardener/gardener-extensions/controllers/provider-aws/pkg/apis/config/helper"
+	"github.com/gardener/gardener-extensions/controllers/provider-aws/pkg/aws"
+	"github.com/gardener/gardener-extensions/pkg/controller/worker"
+	"github.com/gardener/gardener-extensions/pkg/util"
+
+	gardencorev1alpha1 "github.com/gardener/gardener/pkg/apis/core/v1alpha1"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	machinev1alpha1 "github.com/gardener/machine-controller-manager/pkg/apis/machine/v1alpha1"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// MachineClassKind yields the name of the AWS machine class.
+func (w *workerContext) MachineClassKind() string {
+	return "AWSMachineClass"
+}
+
+// MachineClassList yields a newly initialized AWSMachineClassList object.
+func (w *workerContext) MachineClassList() runtime.Object {
+	return &machinev1alpha1.AWSMachineClassList{}
+}
+
+// DeployMachineClasses generates and creates the AWS specific machine classes.
+func (w *workerContext) DeployMachineClasses(ctx context.Context) error {
+	if w.machineClasses == nil {
+		if err := w.generateMachineConfig(ctx); err != nil {
+			return err
+		}
+	}
+	return w.seedChartApplier.ApplyChart(ctx, filepath.Join(aws.InternalChartsPath, "machineclass"), w.worker.Namespace, "machineclass", map[string]interface{}{"machineClasses": w.machineClasses}, nil)
+}
+
+// GenerateMachineDeployments generates the configuration for the desired machine deployments.
+func (w *workerContext) GenerateMachineDeployments(ctx context.Context) (worker.MachineDeployments, error) {
+	if w.machineDeployments == nil {
+		if err := w.generateMachineConfig(ctx); err != nil {
+			return nil, err
+		}
+	}
+	return w.machineDeployments, nil
+}
+
+func (w *workerContext) generateMachineClassSecretData(ctx context.Context) (map[string][]byte, error) {
+	secret := &corev1.Secret{}
+	if err := w.client.Get(ctx, kutil.Key(w.worker.Spec.SecretRef.Namespace, w.worker.Spec.SecretRef.Name), secret); err != nil {
+		return nil, err
+	}
+
+	return map[string][]byte{
+		machinev1alpha1.AWSAccessKeyID:     secret.Data[aws.AccessKeyID],
+		machinev1alpha1.AWSSecretAccessKey: secret.Data[aws.SecretAccessKey],
+	}, nil
+}
+
+func (w *workerContext) generateMachineConfig(ctx context.Context) error {
```

Comment:
Basically, there are already two two functions satisfying the interface, `GenerateMachineDeployments` and `DeployMachineClasses`. However, the computation of the machine classes and machine deployments happens together in this function, true. I would like to keep it like this because every machine class belongs to one machine deployment. If I would split I would have to loop twice and to combine both computations because I later need the `MachineDeployment struct` that contains information about both deployment and class.

I'll add some tests, though.